### ISSUE TITLE
Enhancement: Default alt text option

### ISSUE THREAD

**Issue by locika** [wagtail/wagtail#12660]
### Is your proposal related to a problem?

With ImageBlock there is an alt text field (or decorative image option). If an image has been uploaded most of the times the alt text related to one image is most of the time the same.

### Describe the solution you'd like

Add an optional default alt text field to the uploaded images

### Describe alternatives you've considered

If once the alt text has been filled for one image, that alt text should be copied every time when that image will be used

### Working on this

I would like to work on this issue, but I would like to consult with somebody, who has knowledge from the source code.

Anyone can contribute to this. View our [contributing guidelines](https://docs.wagtail.org/en/latest/contributing/index.html), add a comment to the issue once you’re ready to start.

**Comment by gasman** [comment]
Thanks for the suggestion @locika! I believe the 'description' field on the image is supposed to serve this purpose.

@Chiemezuo @Stormheg @thibaudcolas - I was under the impression that the description field on the image model was meant to prefill the 'alt text' field of ImageBlock when an image was selected, but now I see that feature wasn't implemented as part of #11791. Is that an intended feature that was left out due to lack of time, or was there a deliberate decision not to include it (perhaps to encourage editors to write contextual alt text instead of relying on the generic description)? Can you clarify please?

**Comment by Stormheg** [comment]
@gasman yes pre-filling the contextual alt text from the `description` field was originally part of the GSoC project, but we had to drop it.

Our original plan was to have a button next to the description field that copies over the `description` field of an image into the contextual alt text field.

If I recall correctly, the issue with this approach was a technical one. We couldn't get this button to function. @Chiemezuo did I recall this correctly? Anything to add?

**Comment by Chiemezuo** [comment]
Hi @gasman I'll add some things to what @Stormheg mentioned, but it was completely spot on.

We wanted a "suggestion UI" (A box to display the 'description' and possibly anything else e.g AI-generated suggestions). The idea was that clicking an item on the Suggestion UI would then fill the alt text field with either the generic description or the suggestion. The first version of this was to have a button to copy things over, but the solution was more technical than anticipated and couldn't be fixed within the remaining amount of time I had for the project.
The problem had something to do with the complete Image model data not being available when rendering happens in the editing interface. Something along the lines of Telepath not giving the properties that would be needed to make this happen, in one go.

**Comment by gasman** [comment]
Thanks @Stormheg and @Chiemezuo for the extra detail!

Yep, I see that we currently don't pass the description as part of the result from the chooser modal, so that would need to be added (in [ImageChosenResponseMixin](https://github.com/wagtail/wagtail/blob/23275a4cef4d36bb311abe315eb9ddfc43868e8b/wagtail/images/views/chooser.py#L34)). For consistency we'd probably also want to include it in the bundle of data that a pre-populated AdminChooserWidget can access (in [`get_value_data_from_instance`](https://github.com/wagtail/wagtail/blob/23275a4cef4d36bb311abe315eb9ddfc43868e8b/wagtail/images/widgets.py#L26)), and the widget's state as used when duplicating the block ([`getStateFromHtml`](https://github.com/wagtail/wagtail/blob/23275a4cef4d36bb311abe315eb9ddfc43868e8b/client/src/components/ChooserWidget/ImageChooserWidget.js#L29)), although I don't think that's strictly necessary since we're only using the description at the moment that a new image is picked.

In the absence of a full suggestion UI, would prepopulating the alt text field with the image description be considered a useful first step? I may well have a go at picking that up myself if so...

**Comment by Chiemezuo** [comment]
It potentially could be. However, there might be a tendency for content editors to rely a bit too much on the generic descriptions if it's pre-filled. For some, having to erase the pre-filled data to replace with something else might be an annoyance. Perhaps if this were done when the editor clicks a button instead of automatically, that might be a tad better.
What do you think?

cc @thibaudcolas

**Comment by cnk** [comment]
I really like the automatic prefill one gets when choosing an image in Draftail. I think that is a better UX than a button to copy in the info from default_alt_text. For sites with well curated image libraries, the description is probably pretty good and the main decision is wether or not to use 'is decorative' in this context. But even if the user needs to change the text, it is useful to have the image's description available right at the point where they are editing to act as context.

**Comment by thibaudcolas** [comment]
For sites where implementers know their existing "Description" fields are correctly used as part of their strategy to manage alt text, then I think pre-filling the alt text field like we currently do with rich text is good? And yeah it’s probably a good improvement for all sites until we can create our ideal UI for this.

The ideal UI is documented in [RFC 97 in the "Proposed alt flow" diagram](https://github.com/wagtail/rfcs/blob/main/text/097-alt-text-capabilities.md#proposed-alt-flow). We felt it important to encourage the user to make a decision about alt text in the context where the image is used, so the plan was:

- Pre-fill the alt text field with the description if the image was just uploaded within a chooser on the current page.
- Empty alt text field with a button for the user to "use the description as alt text" if the image was selected from the pre-existing library (see ATAG [B.2.3.3 Save for Reuse](https://www.w3.org/TR/2015/NOTE-IMPLEMENTING-ATAG20-20150924/#sc_b233))

The "use this text" button is also a better pattern in a world where alt text options are proposed by generative AI, here is what we had designed, with the idea that the same pattern could be reused for AI-free websites but with a single "use this" option:

![Alt text options ideal UI](https://github.com/user-attachments/assets/4e996996-3f30-4d9d-913f-50d0a0f74dfe)

--- 

Anyway, for now, if we pre-fill with `default_alt_text` it’s probably a win for adoption of the contextual alt text feature, even if there is a risk of the field ending up with the image’s title. Consistency with the rich text experience clearly a win too.
